### ISSUE TITLE
Trusting TLS offloading proxies

### ISSUE THREAD

**Issue by erbilsilik** [adrianbrs/nest-oidc-provider#9]
**Issue:** Currently, it's not possible to set the proxy for the provider. 

**Error**: _oidc-provider WARNING: x-forwarded-proto header not detected for an https issuer, you must configure your ssl offloading proxy and the provider, see documentation for more details: https://github.com/panva/node-oidc-provider/tree/v7.12.0/docs/README.md#trusting-tls-offloading-proxies_

**Solution Proposal:**

```
private static createOidcProvider(): Provider {
    return {
      provide: oidc.Provider,
      useFactory: async (moduleOptions: OidcModuleOptions): Promise<any> => {
        // Change controller path manually until Nest doesn't provide an official way for this
        // (see https://github.com/nestjs/nest/issues/1438)
        Controller({
          path: validatePath(moduleOptions.path),
          version: moduleOptions.version,
        })(OidcController);

        const provider = new oidc.Provider(
          moduleOptions.issuer,
          moduleOptions.oidc,
        );
        
        -> provider.proxy = true <-
        
        return provider;
      },
      inject: [OIDC_MODULE_OPTIONS],
    };
  }
```

I think that it could be solved by adding `provider.proxy = true` as in the above example or would be better if we can set from nest-oidc-module settings.

**Comment by adrianbrs** [comment]
Hi @erbilsilik, thanks for reporting this issue.
I think it would be better to add an optional `factory` function in the module options, that receives the provider default instance and can return a new instance if needed, or just make changes to the default one. I will add a `proxy` property in the module options too, for convinience.

I will work on this change and release a new version, if you have any other suggestions, feel free to contribute. Thanks!

**Comment by adrianbrs** [comment]
Released [v1.1.0](https://github.com/adrianbrs/nest-oidc-provider/releases/tag/1.1.0) 🎉
See [here](https://github.com/adrianbrs/nest-oidc-provider#trusting-tls-offloading-proxies) how to enable proxy in module options.
Thx.

**Comment by erbilsilik** [comment]
Thank you!
